pkg/validator/validator.go

```diff
@@ -72,32 +78,53 @@ func validateNetmaskLen(ipnet *net.IPNet) error {
 	}
 }
 
-func validateNetworks(nets []string, mapIdx int, summary *model.RoutemapSummary) error {
-	var allErrs error
+// ValidateNetwork takes an string representing a network from the JSON source
+// file and validates it.
+// Returns the parsed IP and IPNet along with a slice of possible errors. If the
+// network can not be parsed, only the error slice will contain values. The
+// correctness of the network depends on the returned error slice having zero
+// elements.
+func ValidateNetwork(network string) (net.IP, *net.IPNet, []error) {
+	var errs []error
+
+	ip, ipnet, err := net.ParseCIDR(network)
+	if err != nil {
+		return nil, nil, []error{errUnparsableNetworkAddr}
+	}
+
+	if err = ValidateProperCIDR(ip, ipnet); err != nil {
+		errs = append(errs, err)
+	}
+	if err = ValidateNetmaskLen(ipnet); err != nil {
+		errs = append(errs, err)
+	}
+
+	return ip, ipnet, errs
```

Comment:
You don't need to return `[]error` here.  You can use `multierr` like this:

```go
func ValidateNetwork(network string) (net.IP, *net.IPNet, error) {  // change signature
    ...
    return ip, ipnet, multierr.Combine(errs...)
}
```

Reply:
Made a little different change. I noticed I was not setting the full error message to the individual errors, and in the ValidateNetworks had to re-expand the errors passed from ValidateNetwork() so we can set the CIDR, index and segment index values.